z3c/dependencychecker/db.py

```diff
@@ -275,3 +292,32 @@ def _get_test_extra(self):
                 return self._extras_requirements[candidate]
 
         return []
+
+    def _filter_metapackages(self, dotted_names, reference=None):
+        """Remove dotted names that are metapackages (if they are used)"""
+        if reference is None:
+            reference = self._requirements
+
+        result = []
+        for single_import in dotted_names:
+            provided_package = False
+            for provided in self.reverse_metapackages:
+                if single_import in provided:
+                    provided_package = provided
```

Comment:
It is more "requirement that provides this package" than "provided package", I think.

So: is `providing_requirement` better?

Reply:
I reworked this bit as I was not really happy with the code, would you mind having another look?